src/guide/agent-functions/defining-agent-functions.rst

```diff
@@ -74,12 +84,128 @@ When using the Run-Time Compiled (RTC) functions, optionally in the C++ API or r
+``Int8``      Would be ``int_8`` type template argument 
+``UInt8``     Would be ``uint_8`` type template argument 
+``Int16``     Would be ``int_16`` type template argument 
+``UInt16``    Would be ``uint_16`` type template argument 
+``Int32``     Would be ``int_32`` type template argument 
+``UInt32``    Would be ``uint_32`` type template argument 
+``Int64``     Would be ``int_64`` type template argument 
+``UInt64``    Would be ``uint_64`` type template argument 
+============= ===============================================
+
+In each case the Type is appended to the end of the API function. E.g. ``getVariableInt("name")`` in Python would be ``getVariable<"int">("name")`` in C++, and ``getVariableFloat("name")`` would be  ``getVariable<"float">("name")``  in C++
+
 
 FLAME GPU Device Functions
 --------------------------
 
-If you wish to define regular functions which can be called from agent functions, you can use the :c:macro:`FLAMEGPU_DEVICE_FUNCTION` macro:
+If you wish to define regular functions which can be called from agent functions, you can use the :c:macro:`FLAMEGPU_DEVICE_FUNCTION` macro (in C++), or the ``@pyflamegpu.device_function`` decorator for pure Python. Pure Python device functions require type annotations for arguments and the function return type using supported types.
+
+====================== =================================================================================
+Type                   Description
+====================== =================================================================================
+``int``                Python built in ``int`` type which will translate transpile to a C++ ``int`` 
+``float``              Python built in ``float`` type which will translate transpile to a C++ ``float`` 
+``numpy.byte``         A numpy ``byte`` type which will transpile to a C++ ``char``
+``numpy.ubyte``        A numpy ``ubyte`` type which will transpile to a C++ ``unsigned char``
+``numpy.short``        A numpy ``short`` type which will transpile to a C++ ``short``
+``numpy.ushort``       A numpy ``ushort`` type which will transpile to a C++ ``unsigned short``
+``numpy.intc``         A numpy ``intc`` type which will transpile to a C++ ``int``
+``numpy.uintc``        A numpy ``uintc`` type which will transpile to a C++ ``unsigned int``
+``numpy.uint``         A numpy ``uint`` type which will transpile to a C++ ``unsigned int``
+``numpy.longlong``     A numpy ``longlong`` type which will transpile to a C++ ``long long``
+``numpy.ulonglong``    A numpy ``ulonglong`` type which will transpile to a C++ ``unsigned long long``
+``numpy.half``         A numpy ``half`` type which will transpile to a C++ ``half``
+``numpy.single``       A numpy ``single`` type which will transpile to a C++ ``float``
+``numpy.double``       A numpy ``double`` type which will transpile to a C++ ``double``
+``numpy.longdouble``   A numpy ``longdouble`` type which will transpile to a C++ ``long double``
```

Comment:
CUDA doesn't support long double afaik.

Reply:
Not yet. I'm included to leave it in as this table covers all the numpy data types. I will add a note to say that it is not currently supported in device code.